src/spyglass/utils/h5_helper_fn.py

```diff
@@ -0,0 +1,141 @@
+                ret[k] = self.unpack_scalar(v)
+            elif isinstance(v, h5py.Group):
+                ret[k] = self.assemble_dict(v)
+            else:
+                ret[k] = v
+        return ret
+
+    def obj_to_dict(self, obj):
+        """Convert an h5py object to a dictionary."""
+        if not obj:
+            return dict(empty=True)
+        if isinstance(obj, (Path, str)) and Path(obj).exists():
+            return self.obj_to_dict(self.open_file(Path(obj)))
+        if isinstance(obj, dict):
+            return {k: self.obj_to_dict(v) for k, v in obj.items()}
+        if isinstance(obj, (float, str, int, h5py.Dataset)):
+            return self.unpack_scalar(obj)
+        if isinstance(obj, h5py.Group):
+            return self.assemble_dict(obj)
+        if isinstance(obj, np.ndarray):
+            return self.numpy_to_dict(obj)
+        if isinstance(obj, bytes):
+            return self.unpack_scalar(obj.decode())
+        if isinstance(obj, (list, tuple)):
+            return {"iterable": self.obj_to_dict(x) for x in obj}
+        return json_loads(obj)
+
+    def open_file(self, path):
+        if path.suffix == ".h5":
+            return h5py.File(path, "r")
+        if path.suffix == ".nwb":
+            return f"pointer to {path}"
+        if path.suffix == ".json":
+            return json_loads(path.read_text())
+        if path.suffix == ".yaml":
+            return yaml_safe_load(path.read_text())
+        if path.suffix in ["npy", "npz"]:
+            return np.load(path)
+        return dict(unrecognized_file_type=path.suffix)
+
+    def numpy_to_dict(self, obj):
+        """Convert a numpy object to a dictionary."""
+        if obj.dtype.names:
+            return {k: self.numpy_to_dict(obj[k]) for k in obj.dtype.names}
+        elif getattr(obj, "ndim", 0) == 1:
+            return obj.tolist()
+        return [self.numpy_to_dict(x) for x in obj]
+
+    def sort_list_of_dicts(self, obj):
+        """Sort a list of dictionaries."""
+        return sorted(
+            obj,
+            key=lambda x: sorted(x.keys() if isinstance(x, dict) else str(x)),
+        )
+
+    def compare_dict_values(self, key, oval, nval, level, iteration):
+        """Compare values of a specific key in two dictionaries."""
+        next_level = f"{level} {key}".replace("kwargs ", "")
+        if isinstance(oval, dict):
+            self.compare_dicts(oval, nval, next_level, iteration + 1)
+        elif isinstance(oval, list):
+            self.compare_lists(oval, nval, next_level, iteration)
+        elif oval != nval:
+            show = f"\n\t{oval} != {nval}"[: self.line_limit]
+            print(f"{level} {iteration}: vals differ for {key}{show}")
+
+    def compare_lists(self, old_list, new_list, level, iteration):
+        """Compare two lists of dictionaries."""
+        old_sorted = self.sort_list_of_dicts(old_list)
+        new_sorted = self.sort_list_of_dicts(new_list)
+        for o, n in zip(old_sorted, new_sorted):
```

Comment:
Would have issues with different length lists as zip only iterates until the end of the shorter list. Don't know if it is a problem here.